pkg/trace/api/otlp.go

```diff
@@ -226,7 +226,7 @@ func (o *OTLPReceiver) ReceiveResourceSpans(rspans ptrace.ResourceSpans, header
 		rattr[k] = v.AsString()
 		return true
 	})
-	hostname, hostok := attributes.HostnameFromAttributes(attr)
+	hostname, hostok := attributes.HostnameFromAttributes(attr, false)
```

Comment:
I think we need to be able to configure this, otherwise it won't work as expected in the `datadogexporter`, like in the PR over there: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10423/files#diff-174a0fc07fd52cf6d95f9ce05162a04af9ac82a346c0384118ab30b5d6a453f3

We'll need to add a new setting [here](https://github.com/DataDog/datadog-agent/blob/albertvaka/trace-agent-evp-intake-proxy-api/pkg/trace/config/config.go#L40) and document it in `config_template.yaml`